torch/utils/data/dataset.py

```diff
@@ -85,3 +88,35 @@ def cummulative_sizes(self):
         warnings.warn("cummulative_sizes attribute is renamed to "
                       "cumulative_sizes", DeprecationWarning, stacklevel=2)
         return self.cumulative_sizes
+
+
+class Subset(Dataset):
+    def __init__(self, dataset, indices):
+        self.dataset = dataset
+        self.indices = indices
+
+    def __getitem__(self, idx):
+        return self.dataset[self.indices[idx]]
+
+    def __len__(self):
+        return len(self.indices)
+
+
+def random_split(dataset, lengths):
+    """
+    Randomply split a dataset into non-overlapping new datasets of given lengths
+    ds
+
+    Arguments:
+        dataset (Dataset): Dataset to be split
+        lengths (iterable): lengths of splits to be produced
+    """
+    if sum(lengths) != len(dataset):
+        raise ValueError("Sum of input lengths does not equal the length of the input dataset!")
+
+    indices = randperm(sum(lengths))
+    splits = []
+    for i in range(len(lengths)):
+        offset = sum(lengths[:i + 1])
+        splits.append(Subset(dataset, indices[offset - lengths[i]:offset]))
```

Comment:
This can be easily converted to a list comprehension:
```python
from torch._utils import _accumulate
return [Subset(dataset, indices[offset - length:offset])
        for offset, length in zip(_accumulate(lengths), lengths)]
```

Reply:
ah nice, didn't realize an implementation of accumulate existed in torch._utils (I was lazy and didn't want to do this myself :p). will use this instead